### PR TITLE
Add ADR wrt duplicate 'accno' (and prevent duplicates)

### LINKED ISSUE #861: Schema doesn't disallow headings and accounts with same name
While the schema doesn't disallow headings with account numbers/accnos equal those found in accounts, especially old code queries the 'chart' view for account IDs, which then break: the query returns multiple rows, where only zero or one are allowed.

### PR DESCRIPTION
**Pull request by ehuelsmann**
Closes #861

### CODE REVIEW

File: doc/adr/0024-chart-of-accounts-schema.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+) INHERITS (coa_translation);
+
+ALTER TABLE coa_translation ADD CHECK NOINHERIT (FALSE);
+```
+
+By changing the above, where the commonality will be stored in the `coa_node`
+table directly, adding additional fields to an `account` table which links
+back to the `coa_node` for the common fields, like:
+
+``sql
```
Comment (neilt) [inline doc/adr/0024-chart-of-accounts-schema.md:58]: This should be ``` instead of ``

File: doc/adr/0024-chart-of-accounts-schema.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+This is, however, worse than the problem being addressed: by adding a record
+to the `account` table for an existing `coa_node` record, the heading is
+transformed from a heading to an account from one moment to another.
+
+Additionally, there's no way to guarantee that accounts are the leafs in
+the account hierarchy tree in this schema.
+
+## Decision
+
+The downsides of current schema outweigh those of the proposed schema, so
```
Comment (neilt) [inline doc/adr/0024-chart-of-accounts-schema.md:81]: Don't you have these backwards.

Do you mean the '... the downsides of proposed schema outweigh those of the current schema so there will be no change ...'

**Review by neilt — CHANGES_REQUESTED** [review]

File: doc/adr/0024-chart-of-accounts-schema.md (hunk 2 above)
Comment (ehuelsmann) [reply doc/adr/0024-chart-of-accounts-schema.md:81]: Ouch. Yes, I mean this the other way around...

File: doc/adr/0024-chart-of-accounts-schema.md (hunk 1 above)
Comment (ehuelsmann) [inline doc/adr/0024-chart-of-accounts-schema.md:58]:
```suggestion
```sql
```

File: doc/adr/0024-chart-of-accounts-schema.md (hunk 2 above)
Comment (ehuelsmann) [inline doc/adr/0024-chart-of-accounts-schema.md:81]:
```suggestion
The downsides of the proposed schema outweigh those of the current schema, so
```

**Review by ehuelsmann — COMMENTED** [review]
Thanks for the comments. Adding proposed changes, which I can incorporate through GitHub.

**Pull request merged**